d2networking/d2client/game_client.go

```diff
@@ -189,7 +204,11 @@ func (g *GameClient) handleSpawnItemPacket(packet d2netpacket.NetPacket) error {
 }
 
 func (g *GameClient) handleMovePlayerPacket(packet d2netpacket.NetPacket) error {
-	movePlayer := packet.PacketData.(d2netpacket.MovePlayerPacket)
+	movePlayer, err := d2netpacket.UnmarshalMovePlayer(packet.PacketData)
+	if err != nil {
+		return err
+	}
+
 	player := g.Players[movePlayer.PlayerID]
```

Comment:
also minor, but we may want to check that `player` exists in these kinds of cases